src/store/zns/index.ts

```diff
@@ -4,7 +4,12 @@ export enum SagaActionTypes {
   UpdateRoute = 'zns/saga/updateRoute',
 }
 
-const setRoute = createAction<string>(SagaActionTypes.UpdateRoute);
+export interface RouteApp {
+  route: string;
+  hasAppChanged: boolean;
+}
```

Comment:
i don't *think* we need to export this as a named type. typescript is really good at inferring types, and i find that if you try to strongly type too many things, it can actually be counterproductive. for this kind of thing, where all we're doing is defining a set of parameters that a function takes, we shouldn't need to pass the type around everywhere. let's either just define it inline `createAction<{ route: string; hasAppChanged: boolean }>` or get rid of the export, and just keep the definition local. either is fine, but the more types that get passed around, the more difficult it becomes to untangle things when you go to make a change.